src/promptflow-core/promptflow/core/_errors.py

```diff
@@ -1,8 +1,79 @@
 # ---------------------------------------------------------
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # ---------------------------------------------------------
+from openai import OpenAIError
+
 from promptflow.exceptions import ErrorTarget, SystemErrorException, UserErrorException
 
+openai_error_code_ref_message = "Error reference: https://platform.openai.com/docs/guides/error-codes/api-errors"
+
+
+def to_openai_error_message(e: Exception) -> str:
+    ex_type = type(e).__name__
+    error_message = str(e)
+    # https://learn.microsoft.com/en-gb/azure/ai-services/openai/reference
+    params_chat_model_cannot_accept = ["best_of", "echo", "logprobs"]
+    if error_message == "<empty message>":
+        msg = "The api key is invalid or revoked. " "You can correct or regenerate the api key of your connection."
+        return f"OpenAI API hits {ex_type}: {msg}"
+    # for models that do not support the `functions` parameter.
+    elif "Unrecognized request argument supplied: functions" in error_message:
+        msg = (
+            "Current model does not support the `functions` parameter. If you are using openai connection, then "
+            "please use gpt-3.5-turbo, gpt-4, gpt-4-32k, gpt-3.5-turbo-0613 or gpt-4-0613. You can refer to "
+            "https://platform.openai.com/docs/guides/gpt/function-calling. If you are using azure openai "
+            "connection, then please first go to your Azure OpenAI resource, deploy model 'gpt-35-turbo' or "
+            "'gpt-4' with version 0613, then go to prompt flow connection page, upgrade connection api version to "
```

Comment:
we might not have a prompt flow connection page in local

Reply:
remove it